### PR TITLE
reflect on LdtkLevel

### PR DESCRIPTION
**Pull request by Hellzbellz123**
derives Reflect and FromReflect on the necessary parts for LdtkLevel and i added too World too but its not used

mostly just used too visually  debug things in egui_inspector, but can be useful elswhere.

i have reflect_ignore on the json values because its external

### CODE REVIEW

**Comment by Hellzbellz123** [comment]
i also added this ontop of the most recent 0.8 branch, if you have more changes locally i can rebase when its released

**Comment by Trouv** [comment]
> i also added this ontop of the most recent 0.8 branch, if you have more changes locally i can rebase when its released

That branch is actually generated by a bot that will update its branch automatically when I merge contributions. Its purpose is to automate semver and maintain a changelog. I should probably make a CONTRIBUTING.md that explains this, sorry about that - but I think merging the commits from that branch here rather than in #197 could cause some problems. So, if you want this to be a part of 0.8, could you remove that 4eb608a603123b705b2a819584852df6ac0e6656 commit completely? Not just revert it in another commit but remove it entirely?

File: src/ldtk/mod.rs [hunk 1]
```diff
@@ -526,7 +526,7 @@ pub struct EnumDefinition {
     pub values: Vec<EnumValueDefinition>,
 }
 
-#[derive(Eq, PartialEq, Debug, Default, Clone, Serialize, Deserialize)]
+#[derive(Eq, PartialEq, Debug, Default, Clone, Serialize, Deserialize, Reflect, FromReflect)]
```
Comment (Trouv) [inline src/ldtk/mod.rs:529]: Most of this file is actually auto-generated by the LDtk project, but we've made a lot of changes here. In order to make it easier to maintain these changes in future releases where the LDtk asset structure changes significantly, I've been keeping a list of all the changes to this file in the module-level docs. Could you add another list item there about how `Reflect` and `FromReflect` have been implemented everywhere?

**Review by Trouv — CHANGES_REQUESTED** [review]

**Comment by Hellzbellz123** [comment]
> > i also added this ontop of the most recent 0.8 branch, if you have more changes locally i can rebase when its released
> 
> That branch is actually generated by a bot that will update its branch automatically when I merge contributions. Its purpose is to automate semver and maintain a changelog. I should probably make a CONTRIBUTING.md that explains this, sorry about that - but I think merging the commits from that branch here rather than in #197 could cause some problems. So, if you want this to be a part of 0.8, could you remove that [4eb608a](https://github.com/Trouv/bevy_ecs_ldtk/commit/4eb608a603123b705b2a819584852df6ac0e6656) commit completely? Not just revert it in another commit but remove it entirely?

outside of deleting this pr im unsure how too do that,

as for the documentation of changes, i noticed that there is already a reflect item in the list,
should i do a new item in the list or add it that?

**Comment by Trouv** [comment]
> outside of deleting this pr im unsure how too do that, not too experienced with git

Try `git rebase -i main` and commenting out the offending commit, if you're comfortable with vim

> as for the documentation of changes, i noticed that there is already a reflect item in the list,
should i do a new item in the list or add it that?

Adding it to the existing line is fine. Actually I have a branch locally that is editing this file heavily so whatever you do I'll probably have to redo it anyway lol, so as long as you put it somewhere I can change it if I need to. I just want it in there so I remember to do that.

**Pull request closed without merging**